# Linear Modeling Using the *F* Distribution

>Name: Brynn O'Toole

## Quiz Questions

### Python Setup

In [4]:
import numpy as np
from scipy import stats
from scipy import special
from pandas import DataFrame
from patsy import dmatrices
import statsmodels.api as sm

In [6]:
import plotly.graph_objects as go
import plotly.express as px
import plotly.figure_factory as ff
import plotly.io as pio
pio.templates.default = 'plotly_white'

In [7]:
# Seed the pseudo-random number generator
np.random.seed(12)

# Sample size
n=100

# Create data
wcc = np.round(np.random.normal(15, 5, n), 0)
crp = (wcc * 2) + np.round(np.random.normal(0, 10, n), 0)

lipase = wcc + crp + np.round(np.random.normal(2, 10, n), 0)

# Pandas dataframe obejct
df = DataFrame(
    {'WCC':wcc,
     'CRP':crp,
     'Lipase':lipase}
)

In [8]:
px.scatter_matrix(
    df,
    title='Scatter plot matrix'
)

In [9]:
# Target vector and design matrix
y, X = dmatrices('Lipase ~ WCC + CRP', data = df)

### Question 1

In [11]:
# create a multiple linear regression model using WCC and CRP as predictors of Lipase
model = sm.OLS(y, X)
print(model.fit().summary())

                            OLS Regression Results                            
Dep. Variable:                 Lipase   R-squared:                       0.764
Model:                            OLS   Adj. R-squared:                  0.759
Method:                 Least Squares   F-statistic:                     157.3
Date:                Tue, 07 Apr 2026   Prob (F-statistic):           3.61e-31
Time:                        12:30:58   Log-Likelihood:                -374.15
No. Observations:                 100   AIC:                             754.3
Df Residuals:                      97   BIC:                             762.1
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      2.3741      3.013      0.788      0.4

### Question 2

The F-statisic is a 157.3 indicating that the variance between groups is substantially higher than the variance within groups. 

### Question 3

* WCC: coef is 1.18 which means 1.2 is added to the variable and there is a postive relationship. The p-value is 0.046 which means we fail to reject the null hypothesis, so there is a statistically significant relationship.
* CRP: coef is 0.54 which means 0.54 is added the variable and there is a postive relationship. The p-value is 0.00 which means we fail to reject the null hypothesis, so there is a statistically significant relationship.

### Question 4

The $R^2$ value is 0.764 so it explains $76.4$% of the variance in the model, so the model is a good fit.